content/algorithms/gossip_sub.md

```diff
@@ -1,21 +1,90 @@
+- `P₃b`: **Mesh Message Delivery Failures** for a topic. This is a sticky parameter that counts the number of mesh message delivery failures. Whenever a peer is pruned with a negative score, the parameter is augmented by the rate deficit at the time of prune. This is intended to keep history of prunes so that a peer that was pruned because of underdelivery cannot quickly get re-grafted into the mesh. The parameter is mixed with negative weight.
+- `P₄`: **Invalid Messages** for a topic. This is the number of invalid messages delivered in the topic. This is intended to penalize peers who transmit invalid messages, according to application specific validation rules. It is mixed with a negative weight.
+- `P₅`: **Application Specific** score. This is the score component assigned to the peer by the application itself, using application specific rules. The weight is positive, but the parameter itself has an arbitrary real value, so that the application can signal misbehaviour with a negative score or gate peers before an application specific handshake is completed.
+- `P₆`: **IP Colocation Factor**. This parameter is a threshold for the number of peers using the same IP address. If the number of peers in the same IP exceeds the threshold, then the value is the square of the surplus, otherwise it is 0. This is intended to make it difficult to carry out sybil attacks by using a small number of IPs. The parameter is mixed with a negative weight.
+- `P₇`: **Behavioural Penalty**. This parameter captures penalties applied for misbehaviour. The parameter has an associated (decaying) counter, which is explicitly incremented by the router on specific events. The value of the parameter is the square of the counter and is mixed with a negative weight.
+
+**Parameter Decay**
+
+The topic parameters are implemented using counters maintained internally by the router whenever an event of interest occurs. The counters _decay_ periodically so that their values are not continuously increasing and ensure that a large positive or negative score isn't sticky for the lifetime of the peer.
+
+The decay interval is configurable by the application, with shorter intervals resulting in faster decay.
+
+Each decaying parameter can have its own decay _factor_, which is a configurable parameter that controls how much the parameter will decay during each decay period.
+
+For more details on the parameter calculation, please refer to the [GossipSub Specification](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md)
+
+
+### GossipSub Mitigation Strategies
```

Comment:
this is coming out of the blue a bit, maybe retitle this section and/or add an introductory paragraph.